iree/hal/local/loaders/wasm_module_loader.c

```diff
@@ -0,0 +1,552 @@
+}
+
+//===----------------------------------------------------------------------===//
+// iree_hal_wasm_executable_t
+//===----------------------------------------------------------------------===//
+
+typedef struct {
+  iree_hal_local_executable_t base;
+
+  // Flatbuffer definition referencing the WASM module memory.
+  iree_DyLibExecutableDef_table_t def;
+
+  wasm_module_t wasm_module;
+  wasm_module_inst_t wasm_module_inst;
+  wasm_exec_env_t wasm_exec_env;
+
+  // Resolved entry points from |wasm_module_inst|.
+  iree_host_size_t entry_fn_count;
+  wasm_function_inst_t entry_fns[];
+} iree_hal_wasm_executable_t;
+
+static const iree_hal_local_executable_vtable_t iree_hal_wasm_executable_vtable;
+
+static iree_status_t iree_hal_wasm_executable_load(
+    iree_hal_wasm_executable_t* executable, iree_allocator_t host_allocator) {
+  RuntimeInitArgs init_args;
+  memset(&init_args, 0, sizeof(RuntimeInitArgs));
+
+  // TODO(scotttodd): use Alloc_With_Allocator and forward the host_allocator
+  init_args.mem_alloc_type = Alloc_With_System_Allocator;
+  // init_args.mem_alloc_option.allocator.malloc_func = NULL;
+  //       (iree_allocator_alloc_fn_t + IREE_ALLOCATION_MODE_ZERO_CONTENTS)
+  // init_args.mem_alloc_option.allocator.realloc_func = NULL;
+  //       (iree_allocator_alloc_fn_t + IREE_ALLOCATION_MODE_TRY_REUSE_EXISTING)
+  // init_args.mem_alloc_option.allocator.free_func = NULL;
+  //       (iree_allocator_free_fn_t)
+
+  // TODO(scotttodd): figure out where names come from...
+  //   * a more specific name including "IREE" or a file name would be nice
+  //   * "env" is load bearing for resolving native symbols:
+  //      (import "env" "fmaf" (func $fmaf (type 0)))
+  init_args.native_module_name = "env";
+
+  // `fmaf` is still used by some models
+  // TODO(scotttodd): enforce no imports, remove `--allow-undefined` linker flag
+  static NativeSymbol native_symbols[] = {
+      {/*symbol=*/"fmaf",
+       /*func_ptr=*/fmaf_impl,
+       /*signature=*/"(fff)f",
+       /*attachment=*/NULL},
+  };
+  init_args.native_symbols = native_symbols;
+  init_args.n_native_symbols = IREE_ARRAYSIZE(native_symbols);
+
+  if (!wasm_runtime_full_init(&init_args)) {
+    return iree_make_status(IREE_STATUS_INTERNAL,
+                            "wasm_runtime_full_init failed");
+  }
+
+  flatbuffers_uint8_vec_t embedded_library_vec =
+      iree_DyLibExecutableDef_library_embedded_get(executable->def);
+  char error_buffer[128];
+  executable->wasm_module = wasm_runtime_load(
+      embedded_library_vec, flatbuffers_uint8_vec_len(embedded_library_vec),
+      error_buffer, sizeof(error_buffer));
+  if (!executable->wasm_module) {
+    return iree_make_status(IREE_STATUS_INTERNAL,
+                            "wasm_runtime_load failed: %s", error_buffer);
+  }
+
+  // TODO(scotttodd): make this configurable
+  uint32_t stack_size = 1073741824;  // 1 GB
+  uint32_t heap_size = 1073741824;   // 1 GB
+  executable->wasm_module_inst =
```

Comment:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/afa1feb1a818a28b92980227b809cb793a6781f5/core/iwasm/include/wasm_export.h#L327-L333


```
 * @param stack_size the default stack size of the module instance when the
 *        exec env's operation stack isn't created by user, e.g. API
 *        wasm_application_execute_main() and wasm_application_execute_func()
 *        create the operation stack internally with the stack size specified
 *        here. And API wasm_runtime_create_exec_env() creates the operation
 *        stack with stack size specified by its parameter, the stack size
 *        specified here is ignored.
```

so could set to 0 since it's ignored?